lib/src/stylesheet_graph.dart

```diff
@@ -237,8 +242,9 @@ class StylesheetNode {
   /// stylesheets those imports refer to.
   ///
   /// This may have `null` values, which indicate failed imports.
-  Map<Uri, StylesheetNode> get upstream => UnmodifiableMapView(_upstream);
-  Map<Uri, StylesheetNode> _upstream;
+  Map<Tuple2<Uri, bool>, StylesheetNode> get upstream =>
```

Comment:
It's plausible that we'll want to expose this class publicly relatively soon, and I really don't want to have to break this API once we remove imports. WDYT about making `upstream` refer only to `@use` and `@import`, and have a separate `upstreamImports` field that only contains imports?

Reply:
Yeah, that makes sense. I went back and forth on whether to key on a tuple or just have two maps while I was writing this. Would it make sense to do the same in `ImportCache` for consistency?

Reply:
Since `ImportCache`'s map isn't part of its public API, I think it's fine to have it stay as tuple-keyed.